File: logger/src/main/java/ru/digipeople/logger/LoggerFactory.java
package ru.digipeople.logger;


/**
 * Factory of {@link Logger}.
 *
 * @author Aleksandr Brazhkin
 */
public class LoggerFactory {

    public static ILoggerFactory DEFAULT_FACTORY = new LogcatLoggerFactory();

    private LoggerFactory() {

    }

    public static Logger getLogger(String name) {
        return DEFAULT_FACTORY.getLogger(name);
    }

    public static Logger getLogger(Class<?> clazz) {
        return DEFAULT_FACTORY.getLogger(clazz);
    }
}

File: logger_kotlin/src/main/java/ru/digipeople/logger/LoggerFactory.kt
package ru.digipeople.logger

import kotlin.reflect.KClass


/**
 * Factory of [Logger].
 *
 * @author Aleksandr Brazhkin
 */
object LoggerFactory {

    var DEFAULT_FACTORY: ILoggerFactory = LogcatLoggerFactory()

    @JvmStatic
    fun getLogger(name: String): Logger {
        return DEFAULT_FACTORY.getLogger(name)
    }

    @JvmStatic
    fun getLogger(clazz: Class<*>): Logger {
        return DEFAULT_FACTORY.getLogger(clazz)
    }

    @JvmStatic
    fun getLogger(kClass: KClass<*>): Logger {
        return DEFAULT_FACTORY.getLogger(kClass)
    }

}
